<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Scale-freeness of networks from rule-based models</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RBMs for GRN inference</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="scdata.html">Single-cell data</a>
</li>
<li>
  <a href="scalefree.html">Scale-free analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Scale-freeness of networks from rule-based
models</h1>

</div>


<div id="data" class="section level2">
<h2>Data</h2>
<ul>
<li>In this part, example gene expression data set included in R.ROSETTA
is used.</li>
</ul>
</div>
<div id="testing-scale-freeness-for-johnson-and-genetic-reducers"
class="section level2">
<h2>Testing scale-freeness for Johnson and Genetic reducers</h2>
<pre class="r"><code>library(R.ROSETTA)
library(fitdistrplus)
library(ggplot2)
library(poweRlaw)
library(tidyverse)

out_jon &lt;- rosetta(autcon)
out_ge &lt;- rosetta(autcon, reducer = &quot;Genetic&quot;, pAdjustMethod = &quot;fdr&quot;)

#### Johnson reducer
vals &lt;- unname(table(unlist(strsplit(out_jon$main$feature[which(out_jon$main$pValue&lt;0.05)], &quot;,&quot;))))
freqs &lt;- table(vals)
  
descdist(vals %&gt;% as.numeric, discrete = FALSE,obs.col = &quot;#FF5D1B&quot;)</code></pre>
<p><img src="scalefree_files/figure-html/block1-1.png" width="672" /></p>
<pre><code>## summary statistics
## ------
## min:  1   max:  22 
## median:  6 
## mean:  7.628571 
## estimated sd:  6.010215 
## estimated skewness:  0.7329735 
## estimated kurtosis:  2.575866</code></pre>
<pre class="r"><code>df1 &lt;- data.frame(degrees = as.numeric(freqs %&gt;% names), freqs = as.numeric(freqs %&gt;% unname))

  ggplot(df1, aes(x = degrees, y = freqs)) +
  geom_point(colour = &quot;black&quot;, size = 5,alpha = 7/10) +
  scale_x_continuous(&quot;Degree&quot;,
                     trans = &quot;log10&quot;) +
  scale_y_continuous(&quot;Frequency&quot;,
                     trans = &quot;log10&quot;) +
  ggtitle(&quot;&quot;) +
  theme_bw()+
  theme(text = element_text(size=20))</code></pre>
<p><img src="scalefree_files/figure-html/block1-2.png" width="672" /></p>
<pre class="r"><code># tests for power law
data_pl &lt;- displ$new(vals)

data_pl$getXmin()</code></pre>
<pre><code>## [1] 1</code></pre>
<pre class="r"><code>data_pl$getPars()</code></pre>
<pre><code>## NULL</code></pre>
<pre class="r"><code>data_pl$setXmin(5)
data_pl$setPars(2)
est = estimate_pars(data_pl)
est = estimate_xmin(data_pl)
data_pl$setXmin(est)

# Data CDF (Cumulative density function)
plot(data_pl, pch = 19, col = &quot;#5f5f5f&quot;, xlab=&quot;&quot;, ylab=&quot;CDF&quot;)
lines(data_pl, col=&quot;#9cd700&quot;, lwd=3)</code></pre>
<p><img src="scalefree_files/figure-html/block1-3.png" width="672" /></p>
<pre class="r"><code># Bivariate scatter plot
bs = bootstrap(data_pl, no_of_sims = 1000, threads = 2)
plot(jitter(bs$bootstraps[, 2], factor = 1.2), bs$bootstraps[, 3], pch = 19, col = &quot;#5f5f5f&quot;, ylab=expression(alpha), xlab=expression(&quot;x&quot;[min]))</code></pre>
<p><img src="scalefree_files/figure-html/block1-4.png" width="672" /></p>
<pre class="r"><code>bs_p = bootstrap_p(data_pl, no_of_sims = 1000, threads = 2)

# P value
bs_p$p</code></pre>
<pre><code>## [1] 0.457</code></pre>
<pre class="r"><code># Goodness of fit
bs_p$gof</code></pre>
<pre><code>## [1] 0.1213908</code></pre>
<pre class="r"><code>#### Genetic reducer
vals &lt;- unname(table(unlist(strsplit(out_ge$main$features[which(out_ge$main$pValue&lt;0.01)], &quot;,&quot;))))
freqs &lt;- table(vals)

descdist(vals %&gt;% as.numeric, discrete = FALSE, obs.col = &quot;#FF5D1B&quot;)</code></pre>
<p><img src="scalefree_files/figure-html/block1-5.png" width="672" /></p>
<pre><code>## summary statistics
## ------
## min:  1   max:  19 
## median:  3 
## mean:  4.333333 
## estimated sd:  4.582576 
## estimated skewness:  1.888856 
## estimated kurtosis:  6.300659</code></pre>
<pre class="r"><code>df1 &lt;- data.frame(degrees = as.numeric(freqs %&gt;% names), freqs = as.numeric(freqs %&gt;% unname))

 ggplot(df1, aes(x = degrees, y = freqs)) +
  geom_point(colour = &quot;black&quot;, size = 5,alpha = 7/10) +
  scale_x_continuous(&quot;Degree&quot;,
                     trans = &quot;log10&quot;) +
  scale_y_continuous(&quot;Frequency&quot;,
                     trans = &quot;log10&quot;) +
  ggtitle(&quot;&quot;) +
  theme_bw()+
  theme(text = element_text(size=18))</code></pre>
<p><img src="scalefree_files/figure-html/block1-6.png" width="672" /></p>
<pre class="r"><code># tests for power law
data_pl &lt;- displ$new(vals)

data_pl$getXmin()</code></pre>
<pre><code>## [1] 1</code></pre>
<pre class="r"><code>data_pl$getPars()</code></pre>
<pre><code>## NULL</code></pre>
<pre class="r"><code>data_pl$setXmin(5)
data_pl$setPars(2)
est = estimate_pars(data_pl)
est = estimate_xmin(data_pl)
data_pl$setXmin(est)

# Data CDF (Cumulative density function)
plot(data_pl, pch = 19, col = &quot;#5f5f5f&quot;, xlab=&quot;&quot;, ylab=&quot;CDF&quot;)
lines(data_pl, col=&quot;#9cd700&quot;, lwd=3)</code></pre>
<p><img src="scalefree_files/figure-html/block1-7.png" width="672" /></p>
<pre class="r"><code># Bivariate scatter plot
bs = bootstrap(data_pl, no_of_sims = 1000, threads = 2)
plot(jitter(bs$bootstraps[, 2], factor = 1.2), bs$bootstraps[, 3], pch = 19, col = &quot;#5f5f5f&quot;, ylab=expression(alpha), xlab=expression(&quot;x&quot;[min]))</code></pre>
<p><img src="scalefree_files/figure-html/block1-8.png" width="672" /></p>
<pre class="r"><code>bs_p = bootstrap_p(data_pl, no_of_sims = 1000, threads = 2)

# P value
bs_p$p</code></pre>
<pre><code>## [1] 0.187</code></pre>
<pre class="r"><code># Goodness of fit
bs_p$gof</code></pre>
<pre><code>## [1] 0.1245143</code></pre>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
