<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-01-11" />

<title>scdata</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RBM extensions</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="scdata.html">Single-cell data</a>
</li>
<li>
  <a href="directionality.html">Directionality</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">scdata</h1>
<h4 class="date">2025-01-11</h4>

</div>


<pre class="r"><code>library(R.ROSETTA)
library(Seurat)
library(biomaRt)
library(AnnotationDbi)
library(org.Mm.eg.db)
library(fitdistrplus)
library(tidyverse)
library(ggplot2)
library(ggpubr)
library(ggplotify)

# single cell data (Lee et al., 2022) https://www.ebi.ac.uk/gxa/sc/experiments/E-MTAB-10371/results?plotType=UMAP&amp;plotOption=20&amp;k=6
path = &#39;/Users/matga374/Downloads/E-MTAB-10371-normalised-files/E-MTAB-10371.aggregated_filtered_normalised_counts.mtx&#39;
expression_matrix &lt;- ReadMtx(
  mtx = path,
  features = paste0(path, &quot;_rows&quot;),
  cells = paste0(path, &quot;_cols&quot;))</code></pre>
<p>genes = <a href="mailto:expression_matrix@Dimnames"
class="email">expression_matrix@Dimnames</a>[1] meta =
read.table(“/Users/matga374/Downloads/ExpDesign-E-MTAB-10371.tsv”, sep =
‘, header = TRUE) symbols = mapIds( org.Mm.eg.db, keys =
as.matrix(unlist(genes)) %&gt;% as.vector, column = ’SYMBOL’, keytype =
‘ENSEMBL’) <a href="mailto:expression_matrix@Dimnames"
class="email">expression_matrix@Dimnames</a>[[1]] &lt;-
paste0(unname(symbols),“-”, names(symbols))</p>
<p>injr &lt;- CreateSeuratObject(counts = expression_matrix, project =
“mm”, min.cells = 3, min.features = 200)</p>
<div id="filtration" class="section level1">
<h1>filtration</h1>
<p>injr &lt;- subset(injr, cells =
which(meta$Sample.Characteristic.post.analysis.well.quality. == “fail”),
invert = TRUE)</p>
<p>injr &lt;- NormalizeData(injr, normalization.method = “LogNormalize”,
scale.factor = 10000) injr &lt;- FindVariableFeatures(injr,
selection.method = “vst”, nfeatures = 2000)</p>
</div>
<div id="identify-the-10-most-highly-variable-genes"
class="section level1">
<h1>Identify the 10 most highly variable genes</h1>
<p>top10 &lt;- head(VariableFeatures(injr), 10)</p>
</div>
<div id="plot-variable-features-with-and-without-labels"
class="section level1">
<h1>plot variable features with and without labels</h1>
<p>plot1 &lt;- VariableFeaturePlot(injr) plot2 &lt;- LabelPoints(plot =
plot1, points = top10, repel = TRUE) plot1 + plot2</p>
<p>all.genes &lt;- rownames(injr) injr &lt;- ScaleData(injr, features =
all.genes)</p>
<p>injr &lt;- RunPCA(injr, features = VariableFeatures(object = injr))
DimPlot(injr, reduction = “pca”) + NoLegend()</p>
<p>injr &lt;- FindNeighbors(injr, dims = 1:10) injr &lt;-
FindClusters(injr, resolution = 0.5)</p>
<p>injr &lt;- RunUMAP(injr, dims = 1:10) injr &lt;- RunTSNE(injr, dims =
1:10) # note that you can set <code>label = TRUE</code> or use the
LabelClusters function to help label # individual clusters</p>
<p>cols = meta<span
class="math inline">\(Factor.Value.injury.[-which(meta\)</span>Sample.Characteristic.post.analysis.well.quality.
== “fail”)]</p>
<p>#levels(cols) = c(“black”,“red”)</p>
<p>p1 &lt;- DimPlot(injr, reduction = “umap”) + labs(color = “cluster”)+
theme(text = element_text(size=18))+ theme(legend.position=“top”)</p>
<p>p2 &lt;- DimPlot(injr, reduction = “umap”, cells.highlight =
meta$Assay[cols != “none”], cols.highlight = “red”, cols = “gray”, order
= TRUE)+ scale_color_manual(labels = c(“Homeostasis”, “Injury”), values
= c(“black”, “#f587a9”)) + theme(text = element_text(size=18))+
theme(legend.position=“top”)</p>
<p>library(dplyr) counts.df &lt;- <a href="mailto:injr@assays"
class="email">injr@assays</a><span
class="math inline">\(RNA@layers\)</span>scale.data %&gt;% as.matrix
%&gt;% t %&gt;% as.data.frame rownames(counts.df) &lt;- <a
href="mailto:injr@assays" class="email">injr@assays</a><span
class="math inline">\(RNA@cells@.Data %&gt;% row.names
colnames(counts.df) &lt;- injr@assays\)</span><a
href="mailto:RNA@features" class="email">RNA@features</a>@.Data %&gt;%
row.names</p>
<p>clusterassignemnts &lt;- data.frame(<a
href="mailto:injr@active.ident" class="email">injr@active.ident</a>)
counts.df &lt;- cbind(counts.df, clusterassignemnts)</p>
<p>#counts.df = counts.df[-which(counts.df<span
class="math inline">\(injr.active.ident == 4),]
#counts.df = counts.df[-which(counts.df\)</span>injr.active.ident ==
5),] #counts.df = counts.df[-which(counts.df$injr.active.ident ==
1),]</p>
<p>outcome = paste0(“cluster_”,counts.df$injr.active.ident %&gt;%
as.character) #outcome[outcome == “cluster_0”] &lt;- “cluster0&amp;1”
#outcome[outcome == “cluster_1”] &lt;- “cluster0&amp;1”</p>
<p>#outcome[outcome == “cluster_2”] &lt;- “cluster2&amp;3”
#outcome[outcome == “cluster_3”] &lt;- “cluster2&amp;3”</p>
<p>counts.df$injr.active.ident &lt;- outcome
colnames(counts.df)[ncol(counts.df)] &lt;- “outcome” counts.df &lt;-
counts.df[,c(head(VariableFeatures(injr), 100), “outcome”)]</p>
<p>gnss &lt;- lapply(strsplit(colnames(counts.df),“-”),function(x) x[1])
%&gt;% unlist gnse &lt;-
lapply(strsplit(colnames(counts.df),“-”),function(x) x[2]) %&gt;%
unlist</p>
<p>gnss[which(unlist(gnss)==“NA”)] &lt;- gnse[which(unlist(gnss)==“NA”)]
colnames(counts.df) &lt;- gnss</p>
<p>saveRDS(counts.df,“//Users//matga374//Desktop//sc_data_lee_et_al.rds”)</p>
<p>out_jon_sc &lt;- rosetta(counts.df, discreteParam = 3, underSample =
T, underSampleNum = 20, JohnsonParam = list(Modulo=TRUE, BRT=FALSE,
BRTprec=0.9, Precompute=FALSE, Approximate=TRUE, Fraction=0.9))
out_jon_sc$quality</p>
<p>out_jon_sc$main
saveRDS(out_jon_sc,“//Users//matga374//Desktop//rrosetta_model_sc_disc3_us20_johnson09.rds”)</p>
<p>out_jon_sc_signif &lt;- out_jon_sc<span
class="math inline">\(main[which(out_jon_sc\)</span>main<span
class="math inline">\(pValue&lt;0.05),]
out_jon_sc_signif_double &lt;-
out_jon_sc_signif[grep(&quot;,&quot;,out_jon_sc_signif\)</span>features),]</p>
<p>require(VisuNet)</p>
<p>vis_out &lt;- visunet(out_jon_sc_signif_double, type = “RDF”) rls =
out_jon_sc<span
class="math inline">\(main[which(out_jon_sc\)</span>main<span
class="math inline">\(pValue&lt;0.05),]
rls = rls[which(rls\)</span>accuracyRHS&gt;0.55),] rls = rls<span
class="math inline">\(features[which(rls\)</span>coverageRHS&gt;0.2)]</p>
<p>vals &lt;- unname(table(unlist(strsplit(rls[grep(“,”,rls)], “,”))))
freqs &lt;- table(vals)</p>
<p>descdist(vals %&gt;% as.numeric, discrete = FALSE,obs.col =
“#FF5D1B”) p5 &lt;- recordPlot()</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
